File: services/s3.js
var AWS = require('aws-sdk');
AWS.config.update({
    "accessKeyId":"xx-xx",
    "secretAccessKey":"xx-xx",
    "region":"ap-south-1"   
});
var s3Bucket = new AWS.S3( { params: {Bucket: 'promotions-material'} } );

function fileUploadViaBase64(file_path,base64string)
{
    // console.log("file_path",file_path);
    // console.log("base64string",base64string);
    buf = Buffer.from(base64string.replace(/^data:image\/\w+;base64,/, ""),'base64')
    var data = {
        Key: file_path, 
        Body: buf,
        ACL: 'public-read',
        ContentEncoding: 'base64',
        ContentType: 'image/jpeg'
    };
    s3Bucket.putObject(data, function(err, data){
        if (err) { 
            return false;
        } else {
            return true;
        }
    });
}

async function deleteKey(file_path)
{
    var params = {  Bucket: 'promotions-material', Key: file_path };
    s3Bucket.deleteObject(params, function(err, data) {
    if (err) 
        return err  
    else 
        return data      
    });
}



async function fileUploadViaMultipart(file_path,content,contentType)
{
    var data = {
        Key: file_path, 
        Body: content,
        ACL: 'public-read',
        ContentType: contentType
    };
    let get = await s3Bucket.putObject(data).promise();
    return get;
}

function deleteFolder(file_path,callback){
    var params = {
      Bucket: 'promotions-material',
      Prefix: file_path
    };
  
    s3Bucket.listObjects(params, function(err, data) {
      if (err) return callback(err);
  
      if (data.Contents.length == 0);
  
      params = {Bucket: 'promotions-material'};
      params.Delete = {Objects:[]};
      
      data.Contents.forEach(function(content) {
        params.Delete.Objects.push({Key: content.Key});
      });
  
      s3Bucket.deleteObjects(params, function(err, data) {
        if (err) return err;
        if (data.IsTruncated) {
            deleteFolder('promotions-material');
        } else {
          callback();
        }
      });
    });
  }

module.exports = 
{
    fileUploadViaBase64,
    deleteKey,
    fileUploadViaMultipart,
    deleteFolder
}

File: apps/services/s3.py
import boto3
import base64

# AWS credentials and region
AWS_ACCESS_KEY = 'AKIASLX7KR5YGDXPCJWV'
AWS_SECRET_KEY = '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'
AWS_REGION = 'ap-south-1'

# Configuring AWS credentials and region
s3 = boto3.client(
    's3',
    aws_access_key_id=AWS_ACCESS_KEY,
    aws_secret_access_key=AWS_SECRET_KEY,
    region_name=AWS_REGION
)

def file_upload_via_base64(file_path, base64string):
    # Decoding base64 string and uploading to S3 bucket
    # buf = base64.b64decode(base64string.split('base64_data b')[1])
    print()
    data = {
        'Bucket':'telegramchatproject',
        'Key': file_path,
        'Body': base64string,
        'ACL': 'public-read',
        'ContentEncoding': 'base64',
        'ContentType': 'image/jpeg'
    }
    try:
        s3.put_object(**data)
        print("Successfully uploaded data to telegramchatproject/" + file_path)
        return True
    except Exception as e:
        print(e)
        return False

def delete_key(file_path):
    # Deleting a specific object from S3 bucket
    try:
        s3.delete_object(Bucket='telegramchatproject', Key=file_path)
        return True
    except Exception as e:
        print(e)
        return False

def file_upload_via_multipart(file_path, content, content_type):
    # Uploading content using multipart to S3 bucket
    data = {
        'Key': file_path,
        'Body': content,
        'ACL': 'public-read',
        'ContentType': content_type
    }
    try:
        response = s3.put_object(**data)
        return response
    except Exception as e:
        print(e)
        return None

def delete_folder(file_path):
    # Deleting a folder and its content from S3 bucket
    try:
        response = s3.list_objects_v2(Bucket='telegramchatproject', Prefix=file_path)
        if 'Contents' in response:
            delete_objects = [{'Key': obj['Key']} for obj in response['Contents']]
            s3.delete_objects(Bucket='telegramchatproject', Delete={'Objects': delete_objects})
            if response['IsTruncated']:
                delete_folder(file_path)
        return True
    except Exception as e:
        print(e)
        return False

# Exporting functions
__all__ = [
    'file_upload_via_base64',
    'delete_key',
    'file_upload_via_multipart',
    'delete_folder'
]